src/Migrations/M053CustomisePostsListSeeAllLink.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace P4\MasterTheme\Migrations;
+
+use P4\MasterTheme\MigrationRecord;
+use P4\MasterTheme\MigrationScript;
+
+/**
+ * Customise "see all" link in Posts List blocks.
+ */
+class M053CustomisePostsListSeeAllLink extends MigrationScript
+{
+    private static ?int $news_page_id = null;
+
+    /**
+     * Perform the actual migration.
+     *
+     * @param MigrationRecord $record Information on the execution, can be used to add logs.
+     * phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter -- interface implementation
+     */
+    public static function execute(MigrationRecord $record): void
+    {
+        $check_is_valid_block = function ($block) {
+            return self::check_is_valid_block($block);
+        };
+
+        $transform_block = function ($block) {
+            return self::transform_block($block);
+        };
+
+        Utils\Functions::execute_block_migration(
+            Utils\Constants::BLOCK_QUERY,
+            $check_is_valid_block,
+            $transform_block,
+        );
+    }
+    // phpcs:enable SlevomatCodingStandard.Functions.UnusedParameter
+
+    /**
+     * Check whether a block is a Posts List block with taxonomy filters.
+     *
+     * @param array $block - A block data array.
+     */
+    private static function check_is_valid_block(array $block): bool
+    {
+        // Check if the News & Stories page is defined in the settings.
+        if (!self::news_page_id()) {
+            return false;
+        }
```

Comment:
🤔 this is just for reflection, but don't we already check if the News and Stories page exist before showing this block on the frontend? Meaning if the class `see-all-link` is present we can assume the _News and Stories_ page exists too and just update the URL from the `core/navigation-link` block attributes without needing to check this right?

Reply:
Yes, we do! But this allows us to abort the migration much quicker, rather than look through all inner blocks if there is a navigation link or not